src/modules/virtual-network-peering/main.tf

```diff
@@ -1,10 +1,19 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "= 2.55.0"
+    }
+  }
+}
+
```

Comment:
Any reason to explicitly indicate the provider here? I think the pattern we've been following is to set this at the root module and have the configurations from /modules/* inherit from their caller.

Reply:
Without it I'm currently getting this warning:

╷
│ Warning: Provider azurerm is undefined
│
│   on main.tf line 136, in module "t3-inbound-peering":
│  136:     azurerm = azurerm.hub
│
│ Module module.t3-inbound-peering does not declare a provider named azurerm.
│ If you wish to specify a provider configuration for the module, add an entry for azurerm in the required_providers
│ block within the module.
╵

If you know some other secret to suppressing it, I can do that instead.  But it appears anytime we're going to pass in more than one configured provider we might need to define this block.

Reply:
Huh, odd. Some things I instantly start to consider: Do we get that warning when sourcing the hub provider for the resource group and virtual network in the root tier-3 module? Do we get that warning when deploying the `virtual-network-peering` with the hub provider in tier-0 through tier-2?

Reply:
The error appears on every deployment that uses network-peering.